docs/design/relationship-shared-library.adoc

```diff
@@ -55,31 +55,29 @@ To create such a task, authors need to have:
-* The Tekton implementation can run in a pure Kubernetes cluster, allowing the test suite to be executed in GitHub Actions. Also the target cluster does not have to be OpenShift, allowing to deploy into EKS for example.
+* The Tekton implementation can run in a pure Kubernetes cluster, allowing the test suite to be executed in GitHub Actions. Also, the target cluster does not have to be OpenShift, allowing to deploy into EKS for example.
 * The artifact approach avoids the need to run all components in the release manager pipeline (speeding things up) while at the same time storing all relevant information which should be useful for GxP/SaMD.
 
-### Con
+=== Con
 
-* Tekton tasks are only customizable via paramters so many people might need to create their own tasks because the platform cannot cover all use cases.
-* Each task is one pod - this cause performance overhead as spinning up pods is a bit slow. Jenkins spins up only one pod and therefore is faster.
+* Tekton tasks are only customizable via parameters so many people might need to create their own tasks because the platform cannot cover all use cases.
+* Each task is one pod - this causes performance overhead as spinning up pods is a bit slow. Jenkins spins up only one pod and therefore is faster.
 * There is no way to install plugins or use the UI to e.g. see test execution trends.
 
-### Limitations
+=== Limitations
 
-* Pipeline users cannot specify task resources. This was possible in Jenkins and also used by many users. See issue https://github.com/opendevstack/ods-pipeline/issues/195. Currently support is not even on the Tekton roadmap. Only workaround: multiple tasks or high defaults. If that does not work, users must create their own copies of the tasks.
-* Pipeline users cannot specify sidecars. This was possible in Jenkins and also used by many users (e.g. to spin up a database for testing). See issue https://github.com/opendevstack/ods-pipeline/issues/135. Currently support is not even on the Tekton roadmap. Only workaround: multiple tasks. If that does not work (e.g. you need more than one sidecar), users must create their own copies of the tasks.
+* Pipeline users cannot specify task resources dynamically without changing the task at hand. This was possible in Jenkins and also used by many users. See issue https://github.com/opendevstack/ods-pipeline/issues/195. Currently, support is not even on the Tekton roadmap. Only workaround: multiple tasks or high defaults.
+* Pipeline users cannot specify sidecars dynamically without changing the task at hand. This was possible in Jenkins and also used by many users (e.g. to spin up a database for testing). See issue https://github.com/opendevstack/ods-pipeline/issues/135. Currently, support is not even on the Tekton roadmap. Only workaround: multiple tasks.
```

Comment:
Thanks for the updates to this file!

I think we can take out the first two limitations: specifying resources and sidecars is possible with this PR :)

Reply:
We were thinking about removing those points as well, but after a small discussion decided to add that this is indeed possible but only on a project-wide scope by changing the task itself, not on a per-repo scope as it was possible with a `Jenkinsfile`.

But we are also happy to remove both of those bullets if you think that this is confusing.

Reply:
Yes I think the new solution is good enough to be a solution for now. We should remove this as it isn't a "blocker" any longer.

I would love to have both pain points mentioned in the ADR though, and in the discussion we should link to the issues / Tekton TEPs.